gcm_toolkit/utils/interface.py

```diff
@@ -487,6 +487,156 @@ def phase_curve(self, phases, spectra=None, filename=None):
+        from petitRADTRANS import nat_cst as nc
+
+        # prepare chemistry input (make them 1D arrays)
+        pres = self.dsi[c["Z"]].values
+        co_ratios = np.ones_like(self.dsi[c["Z"]].values) * co_ratio
+        feh_ratios = np.ones_like(self.dsi[c["Z"]].values) * feh_ratio
+        tlist = [('H2O', 'H2O_HITEMP'),
+                 ('H2O', 'H2O_Exomol_R_1'),
+                 ('CO', 'CO_all_iso_HITEMP'),
+                 ('Na', 'Na_allard'),
+                 ('K', 'K_allard')]
+
+        # check if gravity is given, if not use gcm value
+        if gravity is None:
+            gravity = self.dsi.attrs.get(c["g"]) * 100
+
+        # check if radius is given, if not use gcm value
+        if rplanet is None:
+            rplanet = self.dsi.attrs.get(c["R_p"])
+
+        # check if pressure is given, if not use gcm value
+        if pressure_0 is None:
+            pressure_0 = np.max(pres)
+
+        # Avaraging over terminator regions in longitude space
+        ds_clouds_evening = self.dsi.where((self.dsi['lon'] > 90 - lon_resolution/2) *
+                                           (self.dsi['lon'] < 90 + lon_resolution/2),
+                                           drop=True).mean('lon')
+        ds_clouds_morning = self.dsi.where((self.dsi['lon'] > -90 - lon_resolution/2) *
+                                           (self.dsi['lon'] < -90 + lon_resolution/2),
+                                           drop=True).mean('lon')
+
+        # calculate latitude angle
+        lat_step = 180 / lat_points
+
+        # avarage over latitude space
+        spectra_list = []
+        for lat in range(lat_points):
+            tmp_morning = ds_clouds_morning.where((ds_clouds_evening['lat'] > lat*lat_step-90) *
+                                                  (ds_clouds_evening['lat'] < (lat+1)*lat_step-90),
+                                                  drop=True).mean('lat')
+            tmp_evening = ds_clouds_evening.where((ds_clouds_evening['lat'] > lat*lat_step-90) *
+                                                  (ds_clouds_evening['lat'] < (lat+1)*lat_step-90),
+                                                  drop=True).mean('lat')
+
+            # get temperature profile
+            temp = tmp_morning['T'].values[::-1]
+            # calculate chemistry in mass fractions
+            if mass_frac is None:
+                abus = interpol_abundances(co_ratios, feh_ratios, temp, pres)
+                for key in tlist:
+                    if key[0] in abus:
+                        abus[key[1]] = abus[key[0]]
+            else:
+                abus = mass_frac[0][lat]
+            # calcualte transmision spectra for the morning terminator of current lat point
+            self.prt.calc_transm(temp,
+                                 abus,
+                                 gravity,
+                                 mmw*np.ones_like(temp),
+                                 R_pl=rplanet*100,
```

Comment:
Use `rplanet` here and fix units on the input above. Your docstring says that `rplanet` should be passed in cm anyway

Reply:
agree